hazelcast/src/main/java/com/hazelcast/internal/management/request/ChangeClusterStateRequest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.management.request;
+
+import com.eclipsesource.json.JsonObject;
+import com.hazelcast.cluster.ClusterState;
+import com.hazelcast.core.Cluster;
+import com.hazelcast.internal.management.ManagementCenterService;
+
+import java.io.IOException;
+
+import static com.hazelcast.util.JsonUtil.getString;
+
+/**
+ * Request coming from Management Center for changing the {@link ClusterState}
+ */
+public class ChangeClusterStateRequest implements AsyncConsoleRequest {
+
+    private String state;
+
+    public ChangeClusterStateRequest() {
+    }
+
+    public ChangeClusterStateRequest(String state) {
+        this.state = state;
+    }
+
+    @Override
+    public int getType() {
+        return ConsoleRequestConstants.REQUEST_TYPE_CHANGE_CLUSTER_STATE;
+    }
+
+    @Override
+    public Object readResponse(JsonObject in) throws IOException {
+        return getString(in, "result", "FAIL");
+    }
+
+    @Override
+    public void writeResponse(ManagementCenterService mcs, JsonObject out) throws Exception {
+        JsonObject result = new JsonObject();
+        String resultString = "SUCCESS";
+        Cluster cluster = mcs.getHazelcastInstance().getCluster();
+        try {
+            if (state.equalsIgnoreCase("ACTIVE")) {
+                cluster.changeClusterState(ClusterState.ACTIVE);
+            } else if (state.equalsIgnoreCase("FROZEN")) {
+                cluster.changeClusterState(ClusterState.FROZEN);
+
+            } else if (state.equalsIgnoreCase("PASSIVE")) {
+                cluster.changeClusterState(ClusterState.PASSIVE);
+            }
+        } catch (Exception e) {
+            resultString = e.toString();
+        }
+        result.add("result", resultString);
+        out.add("result", result);
+    }
+
```

Comment:
I would pull out a method that converts the cluster state like this:

```
  @Override
    public void writeResponse(ManagementCenterService mcs, JsonObject out) throws Exception {
        JsonObject result = new JsonObject();
        String resultString = "SUCCESS";
        try {
    	   Cluster cluster = mcs.getHazelcastInstance().getCluster();
           cluster.setClusterState(getClusterState(state));		
        } catch (Exception e) {
            resultString = e.toString();
        }
        result.add("result", resultString);
        out.add("result", result);
    }

	
   private static ClusterState getClusterState(String state){
	if ("ACTIVE".equalsIgnoreCase()) {
           return ClusterState.ACTIVE;
        } else if ("FROZEN".equalsIgnoreCase()) {
              return ClusterState.FROZEN;
        } else if("PASSIVE".equalsIgnoreCase(state)){
               return ClusterState.PASSIVE;
        }else{
           throw new IllegalArgumentExcepton(...);
        }
   }
```
If the state is not found; don't ignore the call, but throw an exception. Because this is a problematic situation.

Also when an exception is thrown, you want to log it. Currently it is gobbled up. This makes debugging a lot more complicated.